### ISSUE TITLE
[stable/prometheus-adapter] External metrics issue

### ISSUE THREAD

**Issue by volatilemolotov** [helm/charts#13716]
**Describe the bug**
When using the chart to deploy the adapter to use with HPA after leaving the default values, in this case *rules.default: true* which implies that the external metrics list is empty (im not trying to use external metrics) the charts still deploys the APIService named *external.metrics.k8s.io/v1beta1* causig me to have `error: unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: the server could not find the requested resource
` when calling kubectl api-resource (and other operations, ambassador wont deploy because of this missing api group

**Version of Helm and Kubernetes**:
kubectl:
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"archive", BuildDate:"2019-03-29T11:10:21Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.7-gke.10", GitCommit:"8d9b8641e72cf7c96efa61421e87f96387242ba1", GitTreeState:"clean", BuildDate:"2019-04-12T22:59:24Z", GoVersion:"go1.10.8b4", Compiler:"gc", Platform:"linux/amd64"}
helm:
Client: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}

**Which chart**:
stable/prometheus-adapter

**What happened**:
Chart deploys a apiservice external.metrics and that api is unreachable causing different issues within the clusters

**What you expected to happen**:
Not to have that apiservice (perhaps if .Values.external == true) 

**How to reproduce it** (as minimally and precisely as possible):
Install stable/prometheus-adapter and try to run ambassador via chart

**Comment by spetrashov** [comment]
I can confirm this issue. Tested with the image `prometheus-adapter:v0.5.0`. When I try to request `GET /apis/external.metrics.k8s.io/v1beta1` on the pod directly I get the following result:
```
< HTTP/2 404
< content-type: application/json
< content-length: 179
< date: Wed, 22 May 2019 14:10:07 GMT
<
{
  "paths": [
    "/apis",
    "/apis/custom.metrics.k8s.io",
    "/apis/custom.metrics.k8s.io/v1beta1",
    "/healthz",
    "/healthz/ping",
    "/metrics",
    "/version"
  ]
```

So it seems that `/apis/external.metrics.k8s.io/v1beta1` path is not available.

**Comment by jieyu** [comment]
I posted a fix in #14389. I tested locally and cannot reproduce the issue after applying the fix.

**Comment by volatilemolotov** [comment]
Confirmed fix in 1.0.3
